dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-by-id.controller/index.ts

```diff
@@ -0,0 +1,85 @@
+import { HttpStatusCode } from 'axios';
+import { Response } from 'express';
+import { FeeRecordEntity, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { getBankNameById } from '../../../../repositories/banks-repo';
+import { UtilisationReportRepo } from '../../../../repositories/utilisation-reports-repo';
+import { UtilisationReportReconciliationItem } from '../../../../types/utilisation-reports';
+import { CustomExpressRequest } from '../../../../types/custom-express-request';
+import { NotFoundError, ApiError } from '../../../../errors';
+
+export type GetUtilisationReportReconciliationByIdRequest = CustomExpressRequest<{
+  params: {
+    reportId: string;
+  };
+}>;
+
+type ResponseBody = UtilisationReportReconciliationItem | string;
+
+/**
+ * TODO FN-1615 this function should extract the relevant details from the fee record entity that are needed for the premium payments screen
+ * @param feeRecord - The fee record entity
+ * @returns The mapped fee record
+ */
+const mapFeeRecordEntityToReconciliationItemFeeRecord = (feeRecord: FeeRecordEntity): UtilisationReportReconciliationItem['feeRecords'][number] => ({
```

Comment:
My assumption was that the mapping folder was intended for shared mapping functions rather than any and all mapping functions and, seeing as these functions are specific only to this endpoint (at least, for now), I wanted to keep them near the controller. I based this approach on the reconciliation summary stuff, where a `helper.ts` and `helper.test.ts` contain the mapping functions, but maybe those also belong in the mapping folder. My guess is that the mapping folder is preferable because you can get more granular testing, which a `helper.ts` file would also achieve, but are there any other reasons you have for preferring it over that approach?